### PR TITLE
Release stuff for 4.0

### PR DESCRIPTION
**Pull request by dstansby**
Various release things for 4.0. I'm opening this as a PR to check that the updates to vendored files don't break anything.

### CODE REVIEW

File: sunpy/extern/parse.py [hunk 1]
```diff
@@ -479,7 +479,7 @@
 from functools import partial
 import logging
 
-__all__ = ['parse', 'search', 'findall', 'with_pattern']
+__all__ = 'parse search findall with_pattern'.split()
```
Comment (nabobalis) [inline sunpy/extern/parse.py:482]: They seriously made this change upstream?

**Review by nabobalis — APPROVED** [review]

File: docs/whatsnew/4.0.rst [hunk 2]
```diff
@@ -15,7 +15,47 @@ SunPy 4.0 also includes a large number of smaller improvements and bug fixes, wh
...
+-  Noah Altunian  *
+-  Rohan Sharma  *
+-  Samriddhi Agarwal
+-  Stuart Mumford
+-  Thomas Braccia  *
+-  Tim Gates  *
+-  Will Barnes
+-  William Russell  *
+-  cbard  *
+-  pre-commit-ci[bot]  *
```
Comment (Cadair) [inline docs/whatsnew/4.0.rst:56]: We should exclude this from the script.
```suggestion
```

File: docs/whatsnew/4.0.rst [hunk 3]
```diff
@@ -15,7 +15,47 @@ SunPy 4.0 also includes a large number of smaller improvements and bug fixes, wh
...
+-  Nabil Freij
+-  Noah Altunian  *
+-  Rohan Sharma  *
+-  Samriddhi Agarwal
+-  Stuart Mumford
+-  Thomas Braccia  *
+-  Tim Gates  *
+-  Will Barnes
+-  William Russell  *
+-  cbard  *
```
Comment (Cadair) [inline docs/whatsnew/4.0.rst:55]: Does this poor person not have a real name :stuck_out_tongue:

**Review by Cadair — APPROVED** [review]

File: docs/whatsnew/4.0.rst (hunk 3 above)
Comment (Cadair) [reply docs/whatsnew/4.0.rst:55]: @cbard Do you have a name you would like us to use for your contributions?

File: docs/whatsnew/4.0.rst (hunk 2 above)
Comment (nabobalis) [reply docs/whatsnew/4.0.rst:56]: I want to see them in every release.

Comment (Cadair) [reply docs/whatsnew/4.0.rst:56]: lol

File: docs/whatsnew/4.0.rst (hunk 3 above)
Comment (cbard) [reply docs/whatsnew/4.0.rst:55]: It is true that my birth certificate doesn't say `cbard`... 😛 

You can put in Chris Bard. Could we also put in Raphael Attie in this list as well (if not already in)? His work was the impetus for my putting in a PR in the first place.

Comment (dstansby) [reply docs/whatsnew/4.0.rst:55]: Name updated, and Raphael added 👍

**Pull request merged**